### PR TITLE
First attempt at a plugins list based on dev list feedback

### PR DESCRIPTION
**Pull request by daniel-beck**
This is the first real attempt to build a plugins list for the initial setup based on community feedback. All feedback that wasn't disputed by someone else was incorporated, with the exception of KK asking for `envinject`. I'll need to reconsider that one.

For the _recommended_ plugins, I had to improvise, as these have not been discussed yet. Despite this I decided to provide a "real" selection as we've received quite confused/negative feedback on the previous list contents, and didn't want that repeated with the recommended plugins list.

As there are currently no container-related plugins, I dropped that part from the _Distributed Builds and Containers_ section name. Additionally, several plugins did not fit the existing categories proposed in the wiki. For these, I introduced the section _Build Features_. It contains plugins such as Rebuild, AnsiColor, SSH Agent, or Throttle Concurrents. It resulted in the _General_ category being empty, so I removed that. Alternative suggestions are welcome.

This list also adds Pipeline Stage View due to its popularity despite being very new, and it essentially being part of the Pipeline "solution". This PR therefore replaces #2089.

[**Please provide plugin selection feedback on the dev list. Thanks!**](https://groups.google.com/d/msg/jenkinsci-dev/84dTygm378g/2TkMmYSVFgAJ)

### CODE REVIEW

**Comment by daniel-beck** [comment]
@kzantow PTAL at the PR build. When you select every available plugin for installation during setup, several fail to install as the Maven Plugin is missing. Shouldn't this be installed as dependency? This does not look like [JENKINS-33296](https://issues.jenkins-ci.org/browse/JENKINS-33296) either, as the plugins fail to start due to a dependency far higher than [1.296](https://github.com/jenkinsci/jenkins/blob/f56ace54d173020724d4f3199359df8a191fa311/core/src/main/java/hudson/ClassicPluginStrategy.java#L325).

**Comment by kzantow** [comment]
@daniel-beck the Maven plugin was bundled, so it sounds like the same issue. Because those are not treated as normal dependencies, the order in which they are installed is not enforced the same way currently. I think correcting [JENKINS-33296](https://issues.jenkins-ci.org/browse/JENKINS-33296) should probably solve what you're seeing.

**Comment by daniel-beck** [comment]
@kzantow 

screen shot

These all declare Maven Plugin dependencies:
https://wiki.jenkins-ci.org/display/JENKINS/Dashboard+View
https://wiki.jenkins-ci.org/display/JENKINS/Checkstyle+Plugin
https://wiki.jenkins-ci.org/display/JENKINS/Cobertura+Plugin
etc.

And yet the Maven Plugin does not get installed as dependency. This is not about implied dependencies to e.g. Mailer Plugin or Matrix Project Plugin, as JENKINS-33296. And FWIW, Maven Integration Plugin is being served by the update site as well, so it's not missing or something either.

File: war/src/main/js/api/plugins.js [hunk 1]
```diff
@@ -27,50 +37,84 @@ exports.recommendedPlugins = [
...
+            { "name": "subversion" },
+            { "name": "teamconcert" },
+            { "name": "tfs" }
         ]
     },
     {
-        "category":"Distributed Builds and Containers",
+        "category":"Distributed Builds",
         "plugins": [
             { "name": "matrix-project" },
```
Comment (svanoort) [inline war/src/main/js/api/plugins.js]: Feels like matrix-project would make more sense under the build features section

**Comment by daniel-beck** [comment]
For the next 2.0 preview, I'd really like users to be able to not suffer from tons of exceptions during setup. While [JENKINS-33296](https://issues.jenkins-ci.org/browse/JENKINS-33296) is unresolved, some plugins that depend (transitively) on Matrix Project or Maven Project had to be removed from the list temporarily. Possibly others as well, that don't fail as obviously, but further investigation does not seem worth it.

**Comment by daniel-beck** [comment]
Merging this now so we have a real list of plugins in the upcoming preview release. Further adjustments towards 2.0 will be discussed on the dev list.

**Pull request merged**